crates/sui-tool/src/commands.rs

```diff
@@ -435,6 +448,42 @@ impl ToolCommand {
     #[allow(clippy::format_in_format_args)]
     pub async fn execute(self, tracing_handle: TracingHandle) -> Result<(), anyhow::Error> {
         match self {
+            ToolCommand::ScanConsensusCommits {
+                db_path,
+                start_commit,
+                end_commit,
+            } => {
+                let rocks_db_store = RocksDBStore::new(&db_path);
+
+                let start_commit = start_commit.unwrap_or(0);
+                let end_commit = end_commit.unwrap_or(u32::MAX);
+
+                let commits = rocks_db_store
+                    .scan_commits(CommitRange::new(start_commit..=end_commit))
+                    .unwrap();
+                println!("found {} consensus commits", commits.len());
+
+                for commit in commits {
+                    let inner = &*commit;
+                    let block_refs = inner.blocks();
+                    let blocks = rocks_db_store.read_blocks(block_refs).unwrap();
+
+                    for block in blocks.iter().flatten() {
+                        let data = block.transactions_data();
+                        println!(
+                            "\"index\": \"{}\", \"leader\": \"{}\", \"blocks\": \"{:#?}\", {} txs",
+                            inner.index(),
+                            inner.leader(),
+                            inner.blocks(),
+                            data.len()
+                        );
+                        for txns in &data {
+                            let tx: ConsensusTransaction = bcs::from_bytes(txns).unwrap();
```

Comment:
To actually use this data, I guess you will have to write custom code in a tmp branch, right?